Merge from upstream v13

Alchez opened the pull request.

barredterra committed:
feat: google drive picker

(cherry picked from commit 25a4eb07575094766111c5cca9b46f79dd1ed59f)

barredterra committed:
fix: scope for google picker

(cherry picked from commit 7d333dbf703001c50dc80aa0a28477b741e9b51f)

barredterra committed:
fix: remove log statements

(cherry picked from commit c44cecae5db1ef7231b742ca84ade406224e17d6)

barredterra committed:
feat: better home view for google drive picker

(cherry picked from commit f9dbbf69e0e89e3f36ac821994cfccb6a576f32d)

barredterra committed:
feat: localize google drive picker

(cherry picked from commit 11f4edf051bf9842e58228faae2d2a349d2e763d)

barredterra committed:
refactor: bind this

(cherry picked from commit 741011085b6ba3cccc86ba3514d9deff4eb82b54)

barredterra committed:
refactor: rename GoogleDrive -> GoogleDrivePicker

(cherry picked from commit a018e7b4458c028ebed29d8e889ae72f086d0479)

barredterra committed:
refactor: save one API call

(cherry picked from commit 144608241e9825af20c88b122a70f29d22aa5a41)

barredterra committed:
fix: spaces to tabs

(cherry picked from commit 4706be34ee861797151923e36f94b464ca85306c)

barredterra committed:
fix: define global vars for sider

(cherry picked from commit ffc4c3e060b2350dec2b9f45acb530bf25b1e365)

barredterra committed:
feat: separate switch for picker, descriptions

(cherry picked from commit eb98f338953bb49718b0d321f01b4f8bb25ac6cb)

barredterra committed:
test: google settings

(cherry picked from commit 7c78fda472e9d7ea8264ed21bfe5b7948d77d531)

barredterra committed:
test: picker enabled

(cherry picked from commit 41c6c88c05cf38f96e4643619ad916585edc942c)

barredterra committed:
feat: use nice file name

(cherry picked from commit 6d523c77e32752ec6d0509ac55032b887716967a)

barredterra committed:
fix: new google drive logo

(cherry picked from commit 41e6aa6972832b86b59a9be4b231e92f1592bb86)

barredterra committed:
refactor: suggestions from review

(cherry picked from commit 37dc0b12ef03b62a1209f70c8d51fd71c0257bf5)

barredterra committed:
feat: links to google cloud console in description

(cherry picked from commit 024b8960cdddc2c631961a902633595d18a8c549)

barredterra committed:
test: change test cases after interface change

(cherry picked from commit 99594b0fd87d802e5b3b93d2e94b1a0b91153a42)

barredterra committed:
test: remove test_picker_as_guest

Guest access was disabled. Testing this would be equivalent to testing frappe.whitelist.

(cherry picked from commit b02d2715be971cfccf4ebaa51a2bc33821d520c5)

barredterra committed:
style: use double quotes

(cherry picked from commit d9044815f4034a51d27e4949ffcd153e253cc882)

barredterra committed:
feat: remove "All" access from Google Settings

(cherry picked from commit 5592bf6b1c210336be13b8281ee71aaa895d0ba8)

shariquerik committed:
fix: Get unique records from backend for list

(cherry picked from commit 62a6f8f4ab8f6864e991f03b62bdb354a757cbb5)

shariquerik committed:
fix: sider fix

(cherry picked from commit aae59f61cd3939e23509659636a9cb504c4e24c4)

rohitwaghchaure committed:
feat: user wise configure grid columns

(cherry picked from commit ddb0bef919867a690c9615f43d4fe44d63c4b78e)

rmehta committed:
feat(minor): Added app_name and app_logo in Website Settings

(cherry picked from commit 70899a7a959be3d7ba26bd6aa6783a7123434c24)

shadrak98 committed:
chore: overriding of modified_by and owner

(cherry picked from commit 564aa90499ea54f4aaf8cbee55a673e407172bd8)

shariquerik committed:
fix: Removed unnessary code to get unique data in base_list

(cherry picked from commit a62578a4b9663a1cb6c6311ee70a6aaa0f773d4d)

shadrak98 committed:
feat: applied rate-limiting on web-forms to avoid bulk submission

(cherry picked from commit f3bc29cbdc19066c14acdfa8b7a47f3dbcc9c73e)

nextchamp-saqib committed:
feat: change custom button type (#14074)

* feat: change custom button type using frm

(cherry picked from commit 62a205fe2efd9763b31e67009edce46fee2cad3f)

# Conflicts:
#	frappe/public/js/frappe/ui/page.js

deepeshgarg007 committed:
feat: Enhancements in Server Script

(cherry picked from commit 55e2af20c66b7bf490ae3c4cfb5d3f4d58e885ca)

deepeshgarg007 committed:
fix: Syntax Error

(cherry picked from commit a1251aace99bef5166925e79926a146f79bd7cb6)

shadrak98 committed:
feat: made key optional and added IP flag

(cherry picked from commit 5cc437aa6b593d822e79588c64c85a21631a3033)

shadrak98 committed:
fix: expiry was not setting on key

(cherry picked from commit 5343b28ab5ee1e555276a63eddc6760c0fbb0e1e)

rohitwaghchaure committed:
fix: code cleanup

(cherry picked from commit 3c0d61f198833238a728e82bfc07f69549839df4)

rohitwaghchaure committed:
fix: columns width validation

(cherry picked from commit c7d474b3a2f79135267da1874573fd0a6a016ebc)

shadrak98 committed:
fix: test cases for web forms

(cherry picked from commit 930dddc558c3e4ff098ed8e873569e66973f044e)

shadrak98 committed:
fix(minor): sider issues

(cherry picked from commit 851778e5616db4c9f7a40fb6bd52514f28a10770)

shadrak98 committed:
fix(minor): code clean up

(cherry picked from commit b057a07259c465afac0ee13727a8dac40b978470)

leela committed:
fix: date mismatches while displaying in short form

(cherry picked from commit dd563001096f957722f708e829cfe5c28a802d21)

deepeshgarg007 committed:
fix: Validate server script for doc events

(cherry picked from commit dffd78d3fca385dc4e5ee536beae93099746cdff)

deepeshgarg007 committed:
fix: Linting Issues

(cherry picked from commit 6ab38e5d65406f927bfee1eeb708ae553a0f962d)

deepeshgarg007 committed:
test: Check for doc event validations and cache methods

(cherry picked from commit e13f99dcb1577a7bfaa26f206be4eb0389c7786f)

nextchamp-saqib committed:
feat: handle timeout and deadlocks in db.sql

(cherry picked from commit e205acae3c7605bae79dd5f661dbd7820572fa85)

# Conflicts:
#	frappe/exceptions.py

nextchamp-saqib committed:
chore: improve message & title

(cherry picked from commit 4109caf1bfee2687ad0d8dbe82cc42f4ee5435c6)

nextchamp-saqib committed:
feat: add on_get_node event in treeview

(cherry picked from commit 2072726865a1937640065352b03e9538c1a2ec8b)

ankush committed:
ci: fail on asset build failure

linuxipho committed:
fix: include 'Duration' in fieldtype to fetch

shariquerik committed:
fix: Leaderboard is not loading for user with language other than en

(cherry picked from commit 8f8f80fef7ef0e61124e0324d3a94d88568c9054)

KrithiRamani committed:
passed filename as argument

(cherry picked from commit 2d018a8591f352e2909335a655c04532e82fa6a9)

KrithiRamani committed:
renamed argument to file_path. Added pycups to requirements.txt

(cherry picked from commit 78a89fdb99e4b64e56a4c212ae12f460019d378b)

KrithiRamani committed:
Update frappe/utils/print_format.py

As suggested, moved new args to the end for supporting non-kwarg function calls.

Co-authored-by: Ankush Menat <ankushmenat@gmail.com>
(cherry picked from commit ad89c1fcd859ec297d9832ccbf2df478f9f60ccf)

KrithiRamani committed:
Update requirements.txt

Added version no for pycups

Co-authored-by: Ankush Menat <ankushmenat@gmail.com>
(cherry picked from commit 0e16fc03ad52e6f55d407644c3246f18bb5f2f5f)

rohitwaghchaure committed:
Merge pull request #14330 from frappe/mergify/bp/version-13-hotfix/pr-14329

feat: pass filename as argument to print_by_server (backport #14329)

shariquerik committed:
fix: Heatmap legend color not set when color theme is set

(cherry picked from commit c06194ce914ec7cfef240c5720f47ca90765d7b2)

rohitwaghchaure committed:
feat: frappe.errprint in the server script for debugging

(cherry picked from commit 5bd9be243761d58c1e0771e06f7af8733c7f7ba4)

rohitwaghchaure committed:
Merge pull request #14341 from frappe/mergify/bp/version-13-hotfix/pr-14334

feat: frappe.errprint in the server script for debugging (backport #14334)

rohitwaghchaure committed:
chore: added test case

(cherry picked from commit d454aaf2f0f9a359975a3a3099de29789bee6e79)

ankush committed:
fix: remove pycups from requirements

(cherry picked from commit 8904a1c7ade1310ea62b63843490457770559285)

deepeshgarg007 committed:
fix: Encapsulate cache methods

(cherry picked from commit 4ebdabda485e19777bb64613d9eb0fefa3c76b27)

surajshetty3416 committed:
Merge pull request #14345 from frappe/mergify/bp/version-13-hotfix/pr-14344

surajshetty3416 committed:
Merge pull request #14346 from frappe/mergify/bp/version-13-hotfix/pr-14332

surajshetty3416 committed:
fix: Use capital M in max to avoid libsass compatibility issue

https://github.com/sass/sass/issues/2849
https://github.com/sass/sass/issues/2849#issuecomment-820538086

surajshetty3416 committed:
Merge pull request #14348 from surajshetty3416/fix-login-style

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into mergify/bp/version-13-hotfix/pr-14327

surajshetty3416 committed:
Merge pull request #14331 from frappe/mergify/bp/version-13-hotfix/pr-14327

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into mergify/bp/version-13-hotfix/pr-14310

pruthvi145 committed:
fix: filter on ace editor field

(cherry picked from commit d4b78e18a53da36aa2c0e708a75f11b10fb9774b)

surajshetty3416 committed:
Merge pull request #14347 from frappe/mergify/bp/version-13-hotfix/pr-14310

surajshetty3416 committed:
Merge pull request #14352 from frappe/mergify/bp/version-13-hotfix/pr-14351

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into mergify/bp/version-13-hotfix/pr-14020

surajshetty3416 committed:
Merge pull request #14343 from frappe/mergify/bp/version-13-hotfix/pr-14020

barredterra committed:
feat: add context for "is (not) set" filter translations

(cherry picked from commit 9b19b41a552042cabfa9c98643cf53469041e24b)

surajshetty3416 committed:
Merge pull request #14354 from frappe/mergify/bp/version-13-hotfix/pr-14130

surajshetty3416 committed:
Merge pull request #14359 from frappe/mergify/bp/version-13-hotfix/pr-14357

gavindsouza committed:
Merge pull request #14318 from ankush/asset_failure

ci: fail the build if asset bundling fails

shariquerik committed:
fix: Resolved redefine object issue on webform load (#14367)

(cherry picked from commit f0e7605b65f145257cc560d8d6d715126929ea44)

sunhoww committed:
feat: expose frappe.db.exists to Server Script

(cherry picked from commit 1d9000fd85346dbba2aae5850e07162bb9b445ef)

# Conflicts:
#	frappe/utils/safe_exec.py

anupamvs committed:
fix: browser printing broken

(cherry picked from commit 34ba19a131765e8110b6e261f20199e93e2929e5)

surajshetty3416 committed:
Merge pull request #14374 from frappe/mergify/bp/version-13-hotfix/pr-14373

anupamvs committed:
fix: network printer settings permission issue

(cherry picked from commit 5387e48a5995039d0aa284da943922a214134b1b)

surajshetty3416 committed:
Merge pull request #14379 from frappe/mergify/bp/version-13-hotfix/pr-14376

surajshetty3416 committed:
Merge pull request #14368 from frappe/mergify/bp/version-13-hotfix/pr-14367

surajshetty3416 committed:
Merge pull request #14381 from frappe/mergify/bp/version-13-hotfix/pr-14277

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into fix-duration-field-fetch

surajshetty3416 committed:
Merge pull request #14321 from linuxipho/fix-duration-field-fetch

mergify[bot] committed:
fix: `convert_dates_to_str` converts everything except `date` objects (#14384)

Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>
(cherry picked from commit 6b91ade73c07dc1c070ed137cf54a29a3e7b0993)

Co-authored-by: Sagar Vora <sagar@resilient.tech>

rohitwaghchaure committed:
fix: custom dialog box not working

(cherry picked from commit f7a8ee7756fb0b684293077e078144d95cd1b66c)

rohitwaghchaure committed:
Merge pull request #14387 from frappe/mergify/bp/version-13-hotfix/pr-14386

fix: custom dialog box not working (backport #14386)

rohitwaghchaure committed:
feat: option to set the width for the multi-select dialog box

(cherry picked from commit 6886036002a7860ddb349eadd8d0428d288a40a6)

rohitwaghchaure committed:
Merge pull request #14389 from frappe/mergify/bp/version-13-hotfix/pr-14388

feat: option to set the width for the multi-select dialog box (backport #14388)

surajshetty3416 committed:
test: Fix flaky test

- Due to improper cleanup one of the test of Energy Point was failing

surajshetty3416 committed:
fix: Resolve conflicts

sagarvora committed:
fix: unable to delete role permissions

(cherry picked from commit a1ddd25dff8a110c297e1210314ed089d9b35433)

deepeshgarg007 committed:
fix: Add rollback to safe_exec

(cherry picked from commit 4f74f9e4e10b565b66b8099246005a595b67aefd)

surajshetty3416 committed:
Merge pull request #14400 from frappe/mergify/bp/version-13-hotfix/pr-14397

deepeshgarg007 committed:
fix: Restrict commit and rollback for doc events

(cherry picked from commit 1127d3d18472ad92afe8a0686b42c98a346a659a)

# Conflicts:
#	frappe/utils/safe_exec.py

deepeshgarg007 committed:
chore: Resolve conflicts

gavindsouza committed:
fix(safe_exec): remove non existent methods

mergify[bot] committed:
Merge pull request #14391 from frappe/mergify/bp/version-13-hotfix/pr-14370

feat: expose frappe.db.exists & frappe.db.count to Server Script (backport #14370)

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into mergify/bp/version-13-hotfix/pr-14140

surajshetty3416 committed:
Merge pull request #14406 from frappe/mergify/bp/version-13-hotfix/pr-14140

barredterra committed:
feat: allow tuple of doctypes as key

(cherry picked from commit 9ac748dc300e44298cb083e185adb6f6c9c38c6c)

Commit:
feat: Added frappe.qb in server script

surajshetty3416 committed:
fix: Push aggregate_on_field to field list
- for proper extraction of child table

(cherry picked from commit 14ae20bdb8435fdaaef399bfeb3b8e1dca05750a)

surajshetty3416 committed:
test: Add test case to validate aggregation
- based on child table field

(cherry picked from commit 8febefae9ca7e6fee649f1b26dd2bf5ee57ab619)

surajshetty3416 committed:
refactor: Use f-string

Co-authored-by: gavin <gavin18d@gmail.com>
(cherry picked from commit ab009803893781c29abdb6ae25b6eff3f3a76ec9)

gavindsouza committed:
Merge pull request #14423 from Aradhya-Tripathi/version-13-hotfix

feat: Added frappe.qb in server script

anupamvs committed:
fix: Customize is not listed in menu item [Print]

(cherry picked from commit f6e83dbcaf02d550976d2ae5930ea5f2a3b61a71)

surajshetty3416 committed:
test: Update aggregation test

(cherry picked from commit 74ec9657da2b26c7f5474c86874de65b4ee8880d)

surajshetty3416 committed:
test: Fix test

(cherry picked from commit 18df0b0f8c65e253885619c6f480ca8a47a412ed)

surajshetty3416 committed:
Merge pull request #14431 from frappe/mergify/bp/version-13-hotfix/pr-14424

gavindsouza committed:
Merge pull request #14430 from frappe/mergify/bp/version-13-hotfix/pr-14429

fix: Customize is not listed in menu item [Print] (backport #14429)

deepeshgarg007 committed:
fix: Loading state for query reports

(cherry picked from commit 096700a77aaf74e66c7e795a42ba69869d41bf96)

surajshetty3416 committed:
fix: Permission check for default print format

(cherry picked from commit 22ffe27217d25a7392cc9cfd0f1517719c048112)

surajshetty3416 committed:
Merge pull request #14450 from frappe/mergify/bp/version-13-hotfix/pr-14449

mergify[bot] committed:
fix: app uninstallation failure if module def link field isn't called "module" (backport #14427) (#14443)

* refactor: split remove_app in smaller functions

(cherry picked from commit ae88fca3efd97e040ace8f9d4d3e078d4366a738)

* fix: fieldname for linkfield hardcoded to "module"

Remove_app assumed that all link fileds linking to "Module Def" are
called "module", this causes uninstall failure and leaves bench in
pseudo-irrecoverable state. (requiring some manual cleanup)

(cherry picked from commit 412aa9f5cc3c3ce008605499d915f0cde00f51e9)

* fix: change order of deletion

Previous order was:
1. Delete DocTypes
2. Delete linked records
3. Drop tables

Now if linked records belonged to deleted doctypes it causes
uninstallation failure.

Changed order to:

1. Delete linked records
2. Delete DocTypes
3. Drop tables

(cherry picked from commit aa372b099001e1fe9e2966c246f62923c098e7bd)

* refactor: change _delete_modules for testability

(cherry picked from commit 70fcac8acf5e8e1dc0708f1794e676e75e9529e2)

* test: remove app unit tests

(cherry picked from commit 35b242fe271b58a54756f6f2a75ce6e8bc110b73)

# Conflicts:
#	frappe/tests/test_commands.py

* fix: resolve conflict

Co-authored-by: Ankush Menat <ankush@iwebnotes.com>

barredterra committed:
test: test_create_custom_fields

(cherry picked from commit 58de6515e7bd7a54f03cfd35267f85860343350f)

Commit:
feat: prepending tab to table name for UPDATE and INSERT

(cherry picked from commit c6542749cdfc715d70785dfc0631566b60caf218)

mergify[bot] committed:
Merge pull request #14458 from frappe/mergify/bp/version-13-hotfix/pr-14454

feat: prepending `tab` to doctypes in query builder (backport #14454)

rohitwaghchaure committed:
Merge branch 'version-13-hotfix' into 'version-13-pre-release'

rohitwaghchaure committed:
Merge pull request #14460 from rohitwaghchaure/merge-13-hotfix-to-pre-release-for-13-13

chore: Merge branch 'version-13-hotfix' into 'version-13-pre-release'

deepeshgarg007 committed:
fix: Rename loader functions

(cherry picked from commit d257adc1302b720016cc3ed830ede379aee4403e)

surajshetty3416 committed:
Merge pull request #14426 from frappe/mergify/bp/version-13-hotfix/pr-14087

sumaiya2908 committed:
refactor: child table layout in the web-form

(cherry picked from commit 4146cd3ccaa0547668204320bcb5b603db063216)

surajshetty3416 committed:
Merge pull request #14474 from frappe/mergify/bp/version-13-hotfix/pr-14473

surajshetty3416 committed:
fix: Check if parent exist in col before getting doc

This fixes following error while downloading auto email report

```
Traceback (most recent call last):
  File "/Users/sps/benches/develop/apps/frappe/frappe/app.py", line 66, in application
    response = frappe.api.handle()
  File "/Users/sps/benches/develop/apps/frappe/frappe/api.py", line 54, in handle
    return frappe.handler.handle()
  File "/Users/sps/benches/develop/apps/frappe/frappe/handler.py", line 31, in handle
    data = execute_cmd(cmd)
  File "/Users/sps/benches/develop/apps/frappe/frappe/handler.py", line 67, in execute_cmd
    return frappe.call(method, **frappe.form_dict)
  File "/Users/sps/benches/develop/apps/frappe/frappe/__init__.py", line 1205, in call
    return fn(*args, **newargs)
  File "/Users/sps/benches/develop/apps/frappe/frappe/email/doctype/auto_email_report/auto_email_report.py", line 196, in download
    data = auto_email_report.get_report_content()
  File "/Users/sps/benches/develop/apps/frappe/frappe/email/doctype/auto_email_report/auto_email_report.py", line 98, in get_report_content
    columns, data = make_links(columns, data)
  File "/Users/sps/benches/develop/apps/frappe/frappe/email/doctype/auto_email_report/auto_email_report.py", line 253, in make_links
    doc = frappe.get_doc(col.parent, doc_name) if doc_name else None
  File "/Users/sps/benches/develop/apps/frappe/frappe/__init__.py", line 883, in get_doc
    doc = frappe.model.document.get_doc(*args, **kwargs)
  File "/Users/sps/benches/develop/apps/frappe/frappe/model/document.py", line 62, in get_doc
    raise ValueError('First non keyword argument must be a string or dict')
ValueError: First non keyword argument must be a string or dict
```

The issue was introduced via https://github.com/frappe/frappe/pull/13340

(cherry picked from commit 5c5c17687284006e7a12dee69cd810e9f8be5fbc)

surajshetty3416 committed:
Merge pull request #14477 from frappe/mergify/bp/version-13-hotfix/pr-14476

surajshetty3416 committed:
fix: Set limit_page_length to get all letterhead

`limit` is not a valid argument for `frappe.client.get_list`

surajshetty3416 committed:
Merge pull request #14479 from surajshetty3416/fix-listview-printformat-version-13-hotfix

sagarvora committed:
fix: remove duplication in `customize_form.json`

mergify[bot] committed:
fix: set missing `permlevel` for Role Profile (#14486)

(cherry picked from commit c56ded1c4b32adfa16498ffd2d5e8090ed0a78c2)

Co-authored-by: Sagar Vora <sagar@resilient.tech>

gavindsouza committed:
Merge pull request #14495 from gavindsouza/notification-flaky-test

test: Fix flaky test

mergify[bot] committed:
feat: Add enabled to Language (#14494)

(cherry picked from commit 586ef77cd713a2919b8848b114ed7d4ed5e9d1d3)

Co-authored-by: hrwx <himanshuwarekar@yahoo.com>

mergify[bot] committed:
fix: remove/update older whitelist calls (#14496)

(cherry picked from commit c560d67a5709320054b5b265f8f32ece5800ae6e)

Co-authored-by: Sagar Vora <sagar@resilient.tech>

sagarvora committed:
fix: `if_owner` constraint being applied wrongly (#14471)

* fix: `if_owner` constraint being applied wrongly

* test: check if owner on `frappe.get_list`

Co-authored-by: Pruthvi Patel <pruthvipatel145@gmail.com>
(cherry picked from commit 5c86dc80c7905896b9b308754c012afc4576f1c5)

# Conflicts:
#	frappe/model/db_query.py

sagarvora committed:
fix: merge confict

mergify[bot] committed:
fix: handle negative duration (backport #14493) (#14498)

* fix: handle negative duration

(cherry picked from commit 2e4f5e0796a202d5100c5de69ac9e11772029726)

* fix: minor cleanup

(cherry picked from commit 7649cdff4cb21ee62b2c0cc254d7b722d6d11c3b)

Co-authored-by: Raffael Meyer <14891507+barredterra@users.noreply.github.com>
Co-authored-by: Sagar Vora <sagar@resilient.tech>

mergify[bot] committed:
Merge pull request #14485 from resilient-tech/fix-customize-form

fix: remove duplication in `customize_form.json`

mergify[bot] committed:
Merge pull request #14497 from frappe/mergify/bp/version-13-hotfix/pr-14471

fix: `if_owner` constraint being applied wrongly (backport #14471)

shariquerik committed:
fix: On submitting webform with file, record is created but file is not attached in the attachment section (#14328)

nextchamp-saqib committed:
chore: rename exceptions

(cherry picked from commit b11c99137c90bd59ffb0dc49c7be6cb8bb97fa7f)

# Conflicts:
#	frappe/exceptions.py

nextchamp-saqib committed:
chore: better message

(cherry picked from commit 6cd18711d10ae4a27032434606bb674d94044f60)

surajshetty3416 committed:
fix: Return unsorted json in case of type error

(cherry picked from commit c968712450f0369d8f4a30579bf4856528fb6e43)

gavindsouza committed:
refactor(minor): DRY-er human friendly error messages

(cherry picked from commit 67647724e84faaa02f0319e4e24a251ff9baa490)

surajshetty3416 committed:
fix(Print): Select button alignment issue

(cherry picked from commit d47f23dd3a2e088b3a215b9e7998efb1143bbaee)

surajshetty3416 committed:
fix: List header alignment

(cherry picked from commit 7f86137b74ce94bfb9dc24466e5b25bb94923984)

shariquerik committed:
fix: Include Theme for Apps section rendered twice (#14307)

nextchamp-saqib committed:
refactor: handle exception messages on client side

(cherry picked from commit 4f288db878321086f3681ca6bf078ad5dd377d9b)

surajshetty3416 committed:
Merge pull request #14512 from frappe/mergify/bp/version-13-hotfix/pr-14504

nextchamp-saqib committed:
fix: merge conflicts

mergify[bot] committed:
Merge pull request #14507 from frappe/mergify/bp/version-13-pre-release/pr-14074

feat: change custom button type (backport #14074)

surajshetty3416 committed:
Merge pull request #14502 from frappe/webform-file-upload-fix-bp

nextchamp-saqib committed:
fix: prevent report error for AttachmentLimitReached exceptions

(cherry picked from commit 3b4530ef5d93fa41be611569611281e16ada0476)

rohitwaghchaure committed:
chore: change log for v13.13.0

rohitwaghchaure committed:
Merge pull request #14521 from rohitwaghchaure/change-log-for-v13-13

chore: change log for v13.13.0

rohitwaghchaure committed:
bumped to version 13.13.0

rohitwaghchaure committed:
Merge branch 'version-13-pre-release' into version-13

surajshetty3416 committed:
Merge pull request #14525 from frappe/mergify/bp/version-13-hotfix/pr-14419

hasnain2808 committed:
fix: Include Theme for Apps section rendered twice bp v13 (#14510)

fix: Include Theme for Apps section rendered twice bp v13

surajshetty3416 committed:
Merge pull request #14511 from frappe/mergify/bp/version-13-hotfix/pr-13849

mergify[bot] committed:
Merge pull request #14518 from frappe/mergify/bp/version-13-hotfix/pr-14515

fix: prevent report error for AttachmentLimitReached exceptions (backport #14515)

Commit:
Merge remote-tracking branch 'upstream/version-13' into parsimony-production-v13

ghost commented:
### Bug Fixes

* include 'Duration' in fieldtype to fetch ([06ad3b5](https://github.com/ParsimonyGit/frappe/commit/06ad3b534c3131504dcf481be29cb8e1a78739ae))
* Leaderboard is not loading for user with language other than en ([4e87cc4](https://github.com/ParsimonyGit/frappe/commit/4e87cc4ce912c002b603c80f293392bd9f17afa1))
* remove pycups from requirements ([19c9e64](https://github.com/ParsimonyGit/frappe/commit/19c9e64a1d5b033e791cadaf52b8bf6f412a1b7b))
* Heatmap legend color not set when color theme is set ([7462d0a](https://github.com/ParsimonyGit/frappe/commit/7462d0a164304cf536a64fdead933ad230c6b6b5))
* Use capital M in max to avoid libsass compatibility issue ([b389a90](https://github.com/ParsimonyGit/frappe/commit/b389a9023007ed1c1bd865f1fb58d9e6826de84b))
* filter on ace editor field ([7c8a323](https://github.com/ParsimonyGit/frappe/commit/7c8a32374c8247762e1e1032f41bfd30f517b43b))
* code cleanup ([57d45d1](https://github.com/ParsimonyGit/frappe/commit/57d45d1fed1185307a7b0f3ec1fe198d677ac8a7))
* columns width validation ([db3c558](https://github.com/ParsimonyGit/frappe/commit/db3c558fc9d00b467074d9e34ad23484b14334d6))


### Features

* frappe.errprint in the server script for debugging ([4f205a3](https://github.com/ParsimonyGit/frappe/commit/4f205a3ca4ed96f70951538a298efff284faf360))
*  **minor:** Added app_name and app_logo in Website Settings ([91553df](https://github.com/ParsimonyGit/frappe/commit/91553dfddbb24d27c7d0279341502eb63b88931d))
* add on_get_node event in treeview ([81aa6b4](https://github.com/ParsimonyGit/frappe/commit/81aa6b4739690ec8c93b96f939793b6fda3e798d))
* user wise configure grid columns ([4406f6d](https://github.com/ParsimonyGit/frappe/commit/4406f6d27fbbcde5f3a578da8d3ebc36326167ec))


### Continuous Integration

* fail on asset build failure ([ba5fa00](https://github.com/ParsimonyGit/frappe/commit/ba5fa00d5661f6bc9e2b3d8ec396a1c7b60e4e9b))


### Chore

* added test case ([c9a8108](https://github.com/ParsimonyGit/frappe/commit/c9a81088790428434dcf563d1a3c5307eb130f9b))


### Contributors

<a href="https://github.com/linuxipho">linuxipho</a>, <a href="https://github.com/KrithiRamani">KrithiRamani</a>, <a href="https://github.com/rohitwaghchaure">rohitwaghchaure</a>, <a href="https://github.com/shariquerik">shariquerik</a>, <a href="https://github.com/rmehta">rmehta</a>, <a href="https://github.com/ankush">ankush</a>, <a href="https://github.com/surajshetty3416">surajshetty3416</a>, <a href="https://github.com/nextchamp-saqib">nextchamp-saqib</a>, <a href="https://github.com/pruthvi145">pruthvi145</a>

<details>
<summary>Commit-Lint commands</summary>
<br />

You can trigger Commit-Lint actions by commenting on this PR:
- `@Commit-Lint merge patch` will merge dependabot PR on "patch" versions (X.X.Y - Y change)
- `@Commit-Lint merge minor` will merge dependabot PR on "minor" versions (X.Y.Y - Y change)
- `@Commit-Lint merge major` will merge dependabot PR on "major" versions (Y.Y.Y - Y change)
- `@Commit-Lint merge disable` will desactivate merge dependabot PR
- `@Commit-Lint review` will approve dependabot PR
- `@Commit-Lint stop review` will stop approve dependabot PR



</details>

shariquerik committed:
fix: In info timeline content user should be owner not comment_email (#14519)

shariquerik committed:
revert: Reverting single vertical scrollbar fix bp v13 (#14528)

surajshetty3416 committed:
Merge pull request #14529 from frappe/info-timeline-user-fix-bp

ankush committed:
fix: remove debug statement

(cherry picked from commit a4772d79231f551a016c60bdfc58da962892ff4f)

surajshetty3416 committed:
fix: Set limit_page_length to get all letterhead

`limit` is not a valid argument for `frappe.client.get_list`

surajshetty3416 committed:
Merge branch 'version-13-hotfix' of https://github.com/frappe/frappe into fix-listview-printformat-version-13-hotfix

mergify[bot] committed:
fix: ignore `flags` in `frappe.client.bulk_update` (backport #14488)

* fix: ignore `flags` in `frappe.client.bulk_update`

* fix: merge conflict

Co-authored-by: Sagar Vora <sagar@resilient.tech>

mergify[bot] committed:
feat: `setDefault` for JS Objects (#14535)

(cherry picked from commit 551f68d85a4eff50837fdae4f1e9760add78489a)

Co-authored-by: Sagar Vora <sagar@resilient.tech>

ankush committed:
fix: debounce awesome_bar search

When system language is not english each call to search translates all
doctypes, reports etc to match them. This is computationally intensive
and freezes DOM while typing.

Debouncing to make it usable for now.

Ideal solution: cache translations of standard objects somehow / debounce + async.

refer: https://github.com/frappe/frappe/issues/13914

- replaced manual implementation of debouncing with frappe.utils.debounce
- increased timeout to 500ms.
- assuming ~30 wpm and 5 letter / word ~= 0.4 second / letter.

(cherry picked from commit 128728a0c295f4df8ec3d568e423bf9b2795f374)

surajshetty3416 committed:
Merge pull request #14541 from frappe/mergify/bp/version-13-hotfix/pr-14538

umairsy committed:
fix: Changed default for Email Log from 90 to 30 days (#14543)

(cherry picked from commit 616d97ce78d9ed41dc2ec8d3819f1e86461b9060)

mergify[bot] committed:
Merge pull request #14544 from frappe/mergify/bp/version-13-hotfix/pr-14531

fix: remove debug statements (backport #14531)

mergify[bot] committed:
refactor: improved design for fetching values using `add_fetch` (backport #14445)

* refactor: improved design for fetching values using `add_fetch`

(cherry picked from commit f2b319920ce17143159fd2c5232b2867038b5887)

* test: add UI test to ensure `fetch_from` works

(cherry picked from commit 7aa8059b579cfdaab0f1a1d750cb38e3e1975ad4)

* fix: merge conflict

* style: apply sider fixes

Co-authored-by: Sagar Vora <sagar@resilient.tech>

mergify[bot] committed:
fix: do not pop `doctype` and `docname` (#14557)

(cherry picked from commit b60f0772dbd9bb3304b15fb98a4cd81698c97a47)

Co-authored-by: Sagar Vora <sagar@resilient.tech>

mergify[bot] committed:
test: fix flaky awesome_bar test (backport #14546) (#14549)

* test: fix flaky awesome_bar test

(cherry picked from commit 5996d6829dff88400b16c4b3bb747188d9834a86)

# Conflicts:
#	cypress/support/commands.js

* fix: resolve conflicts

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
ci: move semgrep rules out of repo (backport #14550) (#14559)

* ci: move semgrep rules out of repo

(cherry picked from commit 488cb3126505a50aa21a064e95e022ca7a3ae6c0)

# Conflicts:
#	.github/helper/semgrep_rules/frappe_correctness.yml
#	.github/helper/semgrep_rules/translate.yml

* chore: change repo name

(cherry picked from commit 614423e41c1c97fd434d550250ba029c4bd58f62)

* fix: delete conflicting files

Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
refactor: move max_writes constant to a class var (#14561)

(cherry picked from commit a39e4290050e18544a53b88d65d33d0fec2193cf)

Co-authored-by: Ankush Menat <ankush@iwebnotes.com>

mergify[bot] committed:
feat: allow fuzzy search in website search (backport #14463) (#14560)

* fix(ux): allow fuzzy search in website search

FuzzyTerm allows edit distance based fuzzy searching. This means "webho" or
"wabhool" will match "webhook".

(cherry picked from commit cbe673255bb1900a6a6c4900f9810d8e0f269839)

# Conflicts:
#	frappe/search/full_text_search.py

* test: make global search tests less flaky

(cherry picked from commit 56f5b3d1f32af727e574159be4e325ff663cd32d)

* fix: dont hardcode search fields

FullTextSearch uses name and content
WebsiteSearch uses title and content

Tests were failing because of hardcoded fieldnames which can't be
overridden by inheriting class without rewriting search function.

Made a separate function for defining search fields.

(cherry picked from commit 4ef4ecdf01945a0f38c77dd98a525bc35079f03e)

* feat: apply boosts on search queries

This gives higher priority to title and match over content.

(cherry picked from commit 99a6874affd1fd61ae093be3f6f72920d317babc)

* fix: resolve conflict

Co-authored-by: Ankush Menat <ankush@iwebnotes.com>
Co-authored-by: Ankush Menat <ankush@frappe.io>

mergify[bot] committed:
fix: Use whoosh AsyncWriter to prevent write locks (backport #14215) (#14562)

* fix: Use whoosh AsyncWriter to prevent write locks

(cherry picked from commit a298566d134ac6476558df1fba1eecb97ebf9914)

* Remove duplicate import

Co-authored-by: ChillarAnand <anand21nanda@gmail.com>
Co-authored-by: Chillar Anand <chillar@avilpage.com>

nextchamp-saqib committed:
fix: sider issues

(cherry picked from commit 248f97f99be4713449cc5fd8d2304074acaf52fd)

nextchamp-saqib committed:
refactor:  extend Exception instead of ValidationError

(cherry picked from commit 4670207dbd7dcdefa0ecf13cd2f48b2ed0d5f981)

# Conflicts:
#	frappe/exceptions.py

gavindsouza committed:
Merge pull request #14500 from frappe/mergify/bp/version-13-hotfix/pr-14434

fix: Loading state for query reports (backport #14434)

gavindsouza committed:
fix: Update modified timestamp for log settings changes

sagarvora committed:
fix: set maxlength property for Link fields

(cherry picked from commit 5fded165e29da51877569cab6724b77a286bd1aa)

# Conflicts:
#	frappe/public/js/frappe/form/controls/data.js

surajshetty3416 committed:
Merge pull request #14545 from frappe/mergify/bp/version-13-hotfix/pr-14543

mergify[bot] committed:
Merge pull request #14537 from frappe/mergify/bp/version-13-hotfix/pr-12715

feat: Google Drive Picker (backport #12715)

gavindsouza committed:
fix: errprint only if traceback?

(cherry picked from commit 125233315f9d898fb4608f6fff2544cd04f3e439)

ankush committed:
fix: allow submitting comment with images only

(cherry picked from commit 866212333d0f19bdfbc04e9fc48af50e17d24a08)

shariquerik committed:
fix: Scrollbar is hidden when search modal is minimised bp v13 (#14508)

Co-authored-by: mergify[bot] <37929162+mergify[bot]@users.noreply.github.com>

mergify[bot] committed:
test: Cypress UI test fixes (backport #14503) (#14587)

Co-authored-by: Shariq Ansari <sharique.rik@gmail.com>

sagarvora committed:
fix: merge conflict

rohitwaghchaure committed:
feat: pdf to base64 convert

(cherry picked from commit 5ef72dd290d8a26bcb1b66510afaa0dd18202dad)

rohitwaghchaure committed:
Merge pull request #14603 from frappe/mergify/bp/version-13-hotfix/pr-14602

feat: pdf to base64 convert (backport #14602)

mergify[bot] committed:
fix(minor): Don't allow changes to global hooks from server scripts (backport #14604) (#14605)

Co-authored-by: Aditya Hase <aditya@adityahase.com>
Co-authored-by: Suraj Shetty <13928957+surajshetty3416@users.noreply.github.com>

mergify[bot] committed:
fix(minor): Don't allow frappe.cache() methods in server scripts (backport #14609) (#14612)

Co-authored-by: Aditya Hase <aditya@adityahase.com>

mergify[bot] committed:
fix: remove mutable defaults (#14597) (#14607)

* fix: mutable defaults

* fix: missing defaults

(cherry picked from commit af4ce8156d75739bc9e8827d81cb882770ef9c4e)

Co-authored-by: Ankush Menat <ankush@frappe.io>

rohitwaghchaure committed:
fix: ignore filename with dotted path  (#14617)

* fix: ignore dotted path in file name

* fix: condition for checking file

* Update data.py

* fix: only allow pdf files

Co-authored-by: Ankush Menat <ankushmenat@gmail.com>
(cherry picked from commit 6ae352a05f298ff3790b84eed51ddc8f61bd4638)

rohitwaghchaure committed:
Merge pull request #14618 from frappe/mergify/bp/version-13-hotfix/pr-14617

fix: ignore filename with dotted path  (backport #14617)

ankush committed:
fix: resolve merge conflicts

ankush committed:
Merge pull request #14589 from frappe/mergify/bp/version-13-hotfix/pr-14588

fix: allow submitting comment with images only (backport #14588)

surajshetty3416 committed:
Merge pull request #14580 from frappe/mergify/bp/version-13-hotfix/pr-14305

surajshetty3416 committed:
Merge branch 'version-13-hotfix' into mergify/bp/version-13-hotfix/pr-14576

ankush committed:
test: enable scheduler back after running the test

(cherry picked from commit 13a2c3403cabbf9391fcd82d6cf97f2c5b8c6b0f)

surajshetty3416 committed:
Merge pull request #14591 from frappe/mergify/bp/version-13-hotfix/pr-14576

mergify[bot] committed:
fix: Changing the drag icon in list settings (backport #14362) (#14509)

Co-authored-by: Shariq Ansari <sharique.rik@gmail.com>
Co-authored-by: Shariq Ansari <30859809+shariquerik@users.noreply.github.com>

ankush committed:
Merge pull request #14799 from frappe/mergify/bp/version-13-hotfix/pr-14798

test: enable scheduler back after running the test (backport #14798)

rohitwaghchaure committed:
chore: merge branch 'version-13-hotfix' into 'version-13-pre-release'

rohitwaghchaure committed:
Merge pull request #14837 from rohitwaghchaure/merge-hotfix-to-pre-release-for-13-14

chore: merge branch 'version-13-hotfix' into 'version-13-pre-release'

mergify[bot] committed:
feat(minor): Implement __getitem__ in Base Document (#14855) (#14867)

* feat(minor): Implement __getitem__ in Base Document

This addition allows for syntactic sugars like writing doc["name"] to
access document name. This addition is solely for the sake of code
simplicity. Since dictionaries in Python follow this standard, we can
write code that works for both Vanilla Dicts, Frappe Dicts & all
Document objects

* fix: Use getattr over get in getitem

For the sake of consistency and some level of sanity in the universe.

Co-authored-by: Ankush Menat <ankush@frappe.io>

* fix: Raise KeyError instead of AttributeError

why: shit breaks down. Specifically stuff in Jinja templates, attributes
that are conditional. Either this is acceptable or subscripting objects
cannot and probably should not be possible along with our pre-condiitons

* chore: Use hasattr instead of get default

Couldn't decide between a style commit or a chore

Co-authored-by: Ankush Menat <ankush@frappe.io>
(cherry picked from commit 583a7ef23923d362c85beeabe57703c2d5b12a65)

Co-authored-by: gavin <gavin18d@gmail.com>

vjFaLk commented:
@Alchez can you look into this single failing test when you get the chance?

revant committed:
ci: migrate docker to github actions (#14865)

Alchez commented:
@vjFaLk, I can't figure out what that test is trying to do, and why it's not raising an exception. And it seems like Frappe is no longer running the test on their dev branch. What do?

rohitwaghchaure committed:
chore: Added change log for version 13.14.0

rohitwaghchaure committed:
Merge pull request #14934 from rohitwaghchaure/change-log-for-v13-14-0

chore: Added change log for version 13.14.0

rohitwaghchaure committed:
bumped to version 13.14.0

rohitwaghchaure committed:
Merge branch 'version-13-pre-release' into version-13

sagarvora committed:
fix: restore `validate_link`

(cherry picked from commit 9503e7788a6578e14cd4fb4f66e2fe8e6ea592b6)

# Conflicts:
#	frappe/public/js/frappe/form/controls/link.js
(cherry picked from commit be3b3b0c4b95806a5a05bc43cecfcba3e868de8c)

mergify[bot] committed:
refactor: minor code cleanup in auto_email_report.py (#14948)

(cherry picked from commit 29785af9158e3e28b435546b869af62df27869d2)

Co-authored-by: abdgad <abdgad@users.noreply.github.com>

mergify[bot] committed:
revert: "feat(minor): Implement __getitem__ in Base Document" (backport #14945) (#14947)

* Revert "feat(minor): Implement __getitem__ in Base Document (#14855)"

This reverts commit 583a7ef23923d362c85beeabe57703c2d5b12a65.

(cherry picked from commit d165fb3d477e65f7a395118f69278f2cb5c7f4e5)

* fix: keyerror in auto email report

Co-authored-by: Ankush Menat <ankushmenat@gmail.com>
Co-authored-by: Ankush Menat <ankush@frappe.io>

sagarvora committed:
fix: stricter validation

(cherry picked from commit af41d9f8a5ffd71cd400ca0e726e304ce825a9e5)
(cherry picked from commit ece322c6f5f188715cfd5a152f9420992f492ce5)

sagarvora committed:
test: intercept `validate_link` instead of `get_value`

(cherry picked from commit 1fd53c43068fb636b1cb22150a0c278d5835a2f4)
(cherry picked from commit d9800cd363fbfe99d8c9a23d5180c0ab15fe1df4)

sagarvora committed:
fix: remove `console.log`

(cherry picked from commit cc5c1532995f6e0089475597d0fe92b4eb539840)
(cherry picked from commit e61b979baede4ae67bc924ed7d54cd91d0cc7e01)

sagarvora committed:
fix: merge conflict

(cherry picked from commit 045e497e0c64cddcd1ac235dd0cb8aa7517de929)

sagarvora committed:
fix: use `get_fetch_map`

(cherry picked from commit c64510b6a649aecbcbb3f43d653d86a2a4d1e7df)

sagarvora committed:
fix: cleaner implementation

(cherry picked from commit cfd3002949d2ab3b40d4b9f51a1dc08485126778)

sagarvora committed:
style: newline after block

rohitwaghchaure committed:
Merge pull request #14966 from frappe/mergify/bp/version-13-pre-release/pr-14954

fix: restore `validate_link` (backport #14943) (backport #14954)

rohitwaghchaure committed:
Merge branch 'version-13-pre-release' into version-13

rohitwaghchaure committed:
bumped to version 13.14.1

Commit:
Merge remote-tracking branch 'upstream/version-13' into parsimony-production-v13

Commit:
fix: add missing API response vkey
